gradio/components/chatbot.py

```diff
@@ -507,6 +508,7 @@ def _postprocess_messages_tuples(self, value: TupleFormat) -> ChatbotDataTuples:
     def _postprocess_message_messages(
         self, message: MessageDict | ChatMessage
     ) -> Message:
+        message = copy.deepcopy(message)
```

Comment:
This is needed because if you yield a sequence of messages into Chatbot and they include files, they will be mutated in-place. This was happening in `gr.ChatInterface` here:

```py
    def example_clicked(
        self, example: SelectData
    ) -> Generator[
        tuple[TupleFormat | list[MessageDict], str | MultimodalPostprocess], None, None
    ]:
        """
        When an example is clicked, the chat history (and saved input) is initially set only
        to the example message. Then, if example caching is enabled, the cached response is loaded
        and added to the chat history as well.
        """
        history = []
        self._append_multimodal_history(example.value, None, history)
        example = self._flatten_example_files(example)
        message = example.value if self.multimodal else example.value["text"]
        yield history, message
        if self.cache_examples:
            history = self.examples_handler.load_from_cache(example.index)[0].root
            yield history, message

```